File: src/cn/yiiguxing/plugin/translate/TranslationComponent.java
package cn.yiiguxing.plugin.translate;

import com.intellij.notification.*;
import com.intellij.openapi.components.AbstractProjectComponent;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.NotNull;

import javax.swing.event.HyperlinkEvent;

public class TranslationComponent extends AbstractProjectComponent {

    private static final String DISPLAY_ID_APP_KEY = "NOTIFICATION_APP_KEY";

    private Settings mSettings;

    public TranslationComponent(Project project) {
        super(project);
    }

    @Override
    public void initComponent() {
        super.initComponent();
        mSettings = Settings.getInstance();
    }

    @Override
    public void projectOpened() {
        if (mSettings.isDisableAppKeyNotification() ||
                !Utils.isEmptyOrBlankString(mSettings.getAppId()) || mSettings.isPrivateKeyConfigured())
            return;

        NotificationGroup group = new NotificationGroup(DISPLAY_ID_APP_KEY, NotificationDisplayType.STICKY_BALLOON,
                true);
        String title = "设置App Key";
        String content = String.format("当前App Key为空或者无效，请设置App Key." +
                        "<br/><br/><a href=\"%s\">设置</a> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" +
                        "<a href=\"%s\">不再提示</a>",
                Constants.HTML_DESCRIPTION_SETTINGS, Constants.HTML_DESCRIPTION_DISABLE);
        Notification notification = group.createNotification(title, content, NotificationType.WARNING,
                new NotificationListener.Adapter() {
                    @Override
                    protected void hyperlinkActivated(@NotNull Notification notification,
                                                      @NotNull HyperlinkEvent hyperlinkEvent) {
                        notification.expire();

                        final String description = hyperlinkEvent.getDescription();
                        if (Constants.HTML_DESCRIPTION_SETTINGS.equals(description)) {
                            TranslationOptionsConfigurable.showSettingsDialog(myProject);
                        } else if (Constants.HTML_DESCRIPTION_DISABLE.equals(description)) {
                            mSettings.setDisableAppKeyNotification(true);
                        }
                    }
                });
        Notifications.Bus.notify(notification, myProject);
    }

}

File: src/main/kotlin/cn/yiiguxing/plugin/translate/TranslationComponent.kt
package cn.yiiguxing.plugin.translate

import cn.yiiguxing.plugin.translate.trans.TKK
import cn.yiiguxing.plugin.translate.trans.YoudaoTranslator
import cn.yiiguxing.plugin.translate.util.show
import com.intellij.notification.*
import com.intellij.openapi.components.AbstractProjectComponent
import com.intellij.openapi.project.Project
import javax.swing.event.HyperlinkEvent

class TranslationComponent(project: Project) : AbstractProjectComponent(project) {

    private lateinit var settings: Settings

    override fun initComponent() {
        super.initComponent()
        settings = Settings.instance
        TKK.update()
    }

    override fun projectOpened() {
        if (settings.isDisableAppKeyNotification
                || (settings.translator == YoudaoTranslator.TRANSLATOR_ID
                && settings.youdaoTranslateSettings.appId.isNotBlank()
                && settings.youdaoTranslateSettings.isAppKeyConfigured)) {
            return
        }

        val group = NotificationGroup(DISPLAY_ID_APP_KEY, NotificationDisplayType.STICKY_BALLOON, true)
        val title = "设置有道App Key"
        val content = "当前有道App Key为空或者无效，请设置有道App Key.<br/><br/>" +
                "<a href=\"$HTML_DESCRIPTION_SETTINGS\">设置</a> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" +
                "<a href=\"$HTML_DESCRIPTION_DISABLE\">不再提示</a>"
        group.createNotification(title, content, NotificationType.WARNING,
                object : NotificationListener.Adapter() {
                    override fun hyperlinkActivated(notification: Notification, hyperlinkEvent: HyperlinkEvent) {
                        notification.expire()
                        when (hyperlinkEvent.description) {
                            HTML_DESCRIPTION_SETTINGS -> OptionsConfigurable.showSettingsDialog(myProject)
                            HTML_DESCRIPTION_DISABLE -> settings.isDisableAppKeyNotification = true
                        }
                    }
                }
        ).show(myProject)
    }

    private companion object {
        private const val DISPLAY_ID_APP_KEY = "NOTIFICATION_APP_KEY"
    }
}
